src/services/study/network-map.js

```diff
@@ -137,3 +146,93 @@ export function fetchAllCountries(studyUuid, currentNodeUuid) {
     console.debug(fetchCountriesUrl);
     return backendFetchJson(fetchCountriesUrl);
 }
+
+function createEquipmentIdentifierList(equipmentType, equipmentList) {
+    return {
+        type: 'IDENTIFIER_LIST',
+        equipmentType: equipmentType,
+        filterEquipmentsAttributes: equipmentList.map((eqId) => {
+            if (eqId?.id) {
+                return { equipmentID: eqId.id };
+            } else {
+                return { equipmentID: eqId };
+            }
+        }),
+    };
+}
+export async function createMapFilter(
+    filter,
+    distDir,
+    studyUuid,
+    currentNodeUuid,
+    selectedEquipements
+) {
+    let equipementList = [];
+    switch (filter.equipmentType) {
+        case EQUIPMENT_TYPES.SUBSTATION:
+        case EQUIPMENT_TYPES.LINE:
+            equipementList = createEquipmentIdentifierList(
+                filter.equipmentType,
+                selectedEquipements
+            );
+            break;
+
+        case EQUIPMENT_TYPES.VOLTAGE_LEVEL:
+        case EQUIPMENT_TYPES.BUS:
+        case EQUIPMENT_TYPES.TWO_WINDINGS_TRANSFORMER:
+        case EQUIPMENT_TYPES.THREE_WINDINGS_TRANSFORMER:
+        case EQUIPMENT_TYPES.BUSBAR_SECTION:
+        case EQUIPMENT_TYPES.GENERATOR:
+        case EQUIPMENT_TYPES.BATTERY:
+        case EQUIPMENT_TYPES.LOAD:
+        case EQUIPMENT_TYPES.SHUNT_COMPENSATOR:
+        case EQUIPMENT_TYPES.DANGLING_LINE:
+        case EQUIPMENT_TYPES.STATIC_VAR_COMPENSATOR:
+        case EQUIPMENT_TYPES.HVDC_CONVERTER_STATION:
+        case EQUIPMENT_TYPES.VSC_CONVERTER_STATION:
+        case EQUIPMENT_TYPES.LCC_CONVERTER_STATION:
+        case EQUIPMENT_TYPES.SWITCH:
+            const substationsIds = selectedEquipements.map(
```

Comment:
Too many tests in theses functions
Pb design ?

Reply:
i improved a bit the code. with the usage of 'default'

i added the tests with diffirents errors so the user can have an explicit error message why the selection fails.
for the others test mostly .length > 0.  is due to the backend that have special case for when we pass empty array